apps/cli/lib/cli.ex

```diff
@@ -37,17 +37,24 @@ defmodule CLI do
         end)
 
         # TODO: Use highest known block as limit?
-
-        Sync.sync_new_blocks(
-          block_provider,
-          block_provider_state,
-          updated_trie,
-          chain,
-          blocktree,
-          current_block.header.number + 1,
-          :infinite,
-          highest_known_block_number
-        )
+        if true do
+          Application.put_env(:ex_wire, :initial_db, db)
```

Comment:
one of the issues with this approach is side effects. Is there a way we could avoid this?
A common approach is to send params down the supervision tree.

Reply:
Yeah, this was a lazy approach / didn't really solve what I wanted, but if we think about the CLI just starting sub-app components instead of running code itself (e.g. like what we're doing here), then it's more confusing to me how we pass arguments.

Basically, if we want to supervise ExWire when running CLI, do we choose which of the following options to configure ExWire:

1. `Application.ensure_all_started(:ex_wire)`, passing configuration by some global since there's no way to pass args in that call.
2. CLI is an application with a supervision tree, in it's tree we call: `CLI.start(...); ... children = [ ... ] ++ ExWire.get_child_specs(chain)` and monitor ExWire ourselves?
3. Start ExWire in such a way that it doesn't need any of this data, but where each process can be configured after it's started to do the work required.

Note: similar issues exist with configuration Ethereumex, which has a supervision tree and loads its state from env.

Thoughts?

Reply:
So yeah, I'm not saying overriding configuration is entirely bad. Though you need to identify top level modules/processes that are allowed to write to it. Also, it's not persisted.

There's a couple of options. One is the above, where you limit yourself where you do that - potentially in the application module or top level task module that starts the top level supervisors and workers, which allows you to pass down the configuration. 

One way you could potentially enforce that is using a ETS table (set to protected) per app where the owner of the table is the top level process. Everyone else just reads.